LegendasTVProvider.cs

```diff
@@ -177,7 +177,13 @@ private IEnumerable<string> FindIds(SubtitleSearchRequest request, CancellationT
                     foreach (var suggestion in suggestions)
                     {
                         var source = suggestion._source;
-                        if (source.id_imdb == imdbId && (request.ContentType == VideoContentType.Movie ? source.tipo == "M" : true))
+                        if (!int.TryParse(source.id_imdb, out var sourceImdb))
+                            break;
+                        if (!int.TryParse(imdbId, out var imdbIdInt))
+                            break;
```

Comment:
Since `imdbIdInt` doesn't change, I think this should be before the loop, maybe with a `yield return null` instead of the `break`

Reply:
I have to test it, I'm not sure about the procedure, I didn't pay attention that the variable won't change its value inside the for loop. I think the safest way is going to compare both: string and integer variables. I'll try that as soon as I can.